virtual_ecosystem/models/soil/pools.py

```diff
@@ -415,20 +421,28 @@ def calculate_all_pool_updates(
             A vector containing net changes to each pool. Order [lmwc, maom].
         """
 
-        # Find temperature and soil moisture values for the topsoil layer
-        soil_water_potential = self.data["matric_potential"][
-            top_soil_layer_index
-        ].to_numpy()
-        soil_temperature = self.data["soil_temperature"][
-            top_soil_layer_index
-        ].to_numpy()
-        soil_moisture = self.data["soil_moisture"][top_soil_layer_index].to_numpy()
+        # Find temperature, soil water potential and soil moisture values for the
+        # microbially active depth
+        soil_water_potential = average_water_potential_over_microbially_active_layers(
+            water_potentials=self.data["matric_potential"],
```

Comment:
Is there a reason why you chose DataArray here and convert to numpy in other places?

Reply:
Is basically that I want to make use of the layer structure for the matric potentials but in the other cases I can just use the `numpy` arrays